keel-slack/src/test/kotlin/com/netflix/spinnaker/keel/slack/callbacks/ManualJudgmentCallbackHandlerTests.kt

```diff
@@ -0,0 +1,121 @@
+package com.netflix.spinnaker.keel.slack.callbacks
+
+import com.netflix.spinnaker.keel.api.constraints.ConstraintState
+import com.netflix.spinnaker.keel.api.constraints.ConstraintStatus
+import com.netflix.spinnaker.keel.clouddriver.ResourceNotFound
+import com.netflix.spinnaker.keel.core.api.parseUID
+import com.netflix.spinnaker.keel.persistence.KeelRepository
+import com.netflix.spinnaker.keel.slack.SlackService
+import com.netflix.spinnaker.kork.exceptions.SystemException
+import com.netflix.spinnaker.time.MutableClock
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import strikt.api.expectThat
+import strikt.api.expectThrows
+import strikt.assertions.isEqualTo
+import java.time.Instant
+import java.time.ZoneId
+
+class ManualJudgmentCallbackHandlerTests : JUnit5Minutests {
+
+  class Fixture {
+    val repository: KeelRepository = mockk()
+    val slackService: SlackService = mockk()
+
+    val clock: MutableClock = MutableClock(
+      Instant.parse("2020-03-25T00:00:00.00Z"),
+      ZoneId.of("UTC")
+    )
+
+    val constraintUid = "01EQW0XKNR5H2NMPJXP020EQXE"
+
+    val slackCallbackResponse =
+      SlackCallbackHandler.SlackCallbackResponse(
+        type = "",
+        user = SlackCallbackHandler.SlackUser(
+          username = "keel",
+          id = "01234",
+          name = "keel-user"
+        ),
+        actions = listOf(SlackCallbackHandler.SlackAction(
+          type = "button",
+          action_id = "$constraintUid:OVERRIDE_PASS:MANUAL_JUDGMENT",
```

Comment:
Nit: this format reads kinda backwards to me. I'd expect `MANUAL_JUDGEMENT:$constraintUid:OVERRIDE_PASS` (so constraint type + contraint uid + status). Also, maybe we could use forward slashes instead of colons to make it more REST-ish? Like `MANUAL_JUDGEMENT/01EQW0XKNR5H2NMPJXP020EQXE/OVERRIDE_PASS`?

Reply:
I'm going to think about the structure a bit more, but I can always refactor this after this PR will get merged.